pkg/solana/client/client.go

```diff
@@ -65,6 +68,26 @@ type Client struct {
 	requestGroup *singleflight.Group
 }
 
+type Head struct {
+	rpc.GetBlockResult
+}
+
+func (h *Head) BlockNumber() int64 {
+	if h.BlockHeight == nil {
+		return 0
+	}
+	//nolint:gosec
+	return int64(*h.BlockHeight)
```

Comment:
what's the actual lint error behind this? would be helpful to document why

Reply:
Added a comment to document this for now